src/cascadia/TerminalApp/TerminalPage.cpp

```diff
@@ -1539,26 +1570,44 @@ namespace winrt::TerminalApp::implementation
     {
         if (auto index{ _GetFocusedTabIndex() })
         {
-            _RemoveTabViewItemByIndex(*index);
+            auto tab{ _tabs.GetAt(*index) };
+            _RemoveTab(tab);
         }
     }
 
     // Method Description:
     // - Close the currently focused pane. If the pane is the last pane in the
     //   tab, the tab will also be closed. This will happen when we handle the
     //   tab's Closed event.
-    void TerminalPage::_CloseFocusedPane()
+    winrt::fire_and_forget TerminalPage::_CloseFocusedPane()
     {
         if (const auto terminalTab{ _GetFocusedTabImpl() })
         {
             _UnZoomIfNeeded();
+
+            if (const auto control{ terminalTab->GetActiveTerminalControl() })
+            {
+                if (control.ReadOnly())
+                {
+                    ContentDialogResult warningResult = co_await _ShowCloseReadOnlyDialog();
+
+                    if (warningResult == ContentDialogResult::Primary)
+                    {
+                        co_return;
+                    }
+                }
+            }
+
+            // There is a theoretical race here: if somehow the active pane changes when
```

Comment:
Should we put this logic inside of `Tab::ClosePane`, to try and mitigate the chances for a race here?

I'm imagining in the post #4472 world, someone could totally `wt -w 1 split-pane` to create a new pane in _this_ window, and this race would be easy to repro

Reply:
I guess we should. I was wondering the ROI :blush:

Reply:
Done